### PR TITLE
test: use insta snapshots to add dot_string coverage

### PR DESCRIPTION
**Pull request by ss2165**

### CODE REVIEW

File: src/hugr/views/tests.rs [hunk 1]
```diff
@@ -38,7 +51,19 @@ fn node_connections() -> Result<(), BuildError> {
...
         ]
     );
     Ok(())
 }
+
+#[rstest]
+fn dot_string(sample_hugr: (Hugr, BuildHandle<DataflowOpID>, BuildHandle<DataflowOpID>)) {
+    let (h, _, _) = sample_hugr;
+
+    insta::assert_yaml_snapshot!(h.dot_string());
```
Comment (aborgna-q) [inline src/hugr/views/tests.rs:68]: This is not yaml, so probably should be `assert_snapshot`? Or is it needed for whitespace stuff?

**Review by aborgna-q — COMMENTED** [review]
I'm not sure if we want to test exact matching of the dot format. It was always meant as debug information, so the output is not stable.

File: src/hugr/views/tests.rs (hunk 1 above)
Comment (ss2165) [reply src/hugr/views/tests.rs:68]: it's used for easier debug, its the snap format

**Comment by ss2165** [comment]
Is unstable output so bad? insta makes life easier there - you just review the change is one you expected and accept it.

**Pull request merged**